Gems/ROS2/Code/Source/VehicleDynamics/VehicleInputsState.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Time/ITime.h>
+#include <type_traits>
+//#include <concepts> //C++20
+
+namespace VehicleDynamics
+{
+    /* C++ 20
+    template<typename T>
+    concept Zeroable = std::integral<T> or std::floating_point<T>; // arithmetic types
+
+    template<typename T>
+    requires Zeroable<T> class InputZeroedOnTimeout
+    */
+
+    //! Inputs with an expiration date - effectively is zero after a certain time since update
+    template<typename T, typename = typename AZStd::enable_if<AZStd::is_arithmetic<T>::value, T>::type>
```

Comment:
`enable_if_t` and `is_arithmetic_v` can be used here.
```suggestion
    template<typename T, typename = AZStd::enable_if_t<AZStd::is_arithmetic_v<T>>>
```